src/test/scala/nl.knaw.dans.easy.multideposit/BlackBoxSpec.scala

```diff
@@ -66,80 +92,159 @@ class BlackBoxSpec extends UnitSpec with BeforeAndAfter with MockFactory with Cu
+            .filterNot(_ contains "bag-info.txt")
+            .filterNot(_ contains "manifest-sha1.txt").toSeq
+      }
+
+      it should "check the files in data/" in {
+        doNotRunOnTravis()
+
+        val dataDir = bag.resolve("data/")
+        dataDir.toFile should exist
+        dataDir.listRecursively().map {
+          case file if Files.isDirectory(file) => file.getFileName.toString + "/"
+          case file => file.getFileName.toString
+        } should contain theSameElementsAs dataContent
+      }
+
+      it should "check the files in metadata/" in {
+        doNotRunOnTravis()
+
+        managed(Files.list(bag.resolve("metadata")))
+          .acquireAndGet(_.iterator().asScala.toList)
+          .map(_.getFileName.toString) should contain only("dataset.xml", "files.xml")
+      }
+
+      it should "check metadata/dataset.xml" in {
+        doNotRunOnTravis()
+
+        def removeElemByName(label: String) = new RuleTransformer(new RewriteRule {
+          override def transform(n: Node): Seq[Node] = {
+            n match {
+              case e: Elem if e.label == label => NodeSeq.Empty
+              case e => e
+            }
+          }
+        })
+
+        val datasetXml = bag.resolve("metadata/dataset.xml")
+        val expDatasetXml = expBag.resolve("metadata/dataset.xml")
+        val datasetTransformer = removeElemByName("available")
+
+        // skipping the available field here
+        datasetTransformer.transform(XML.loadFile(datasetXml.toFile)) should
+          equalTrimmed(datasetTransformer.transform(XML.loadFile(expDatasetXml.toFile)))
+      }
+
+      it should "check metadata/files.xml" in {
+        doNotRunOnTravis()
+
+        val filesXml = bag.resolve("metadata/files.xml")
+        val expFilesXml = expBag.resolve("metadata/files.xml")
+
+        XML.loadFile(filesXml.toFile).toSet should equalTrimmed(XML.loadFile(expFilesXml.toFile).toSet)
+      }
+
+      it should "check deposit.properties" in {
+        doNotRunOnTravis()
+
+        val props = settings.outputDepositDir.resolve(s"allfields-$bagName/deposit.properties")
+        val expProps = expectedOutputDir.resolve(s"input-$bagName/deposit.properties")
+
+        // skipping comment lines, as well as the line with randomized bag-id
+        props.read().lines.toSeq should contain allElementsOf
+          expProps.read().lines
+            .filterNot(_ startsWith "#")
+            .filterNot(_ contains "bag-store.bag-id")
+            .toSeq
+      }
     }
+
+    "ruimtereis01" should behave like bagContents("ruimtereis01", expectedDataContentRuimtereis01)
+    "ruimtereis02" should behave like bagContents("ruimtereis02", expectedDataContentRuimtereis02)
+    "ruimtereis03" should behave like bagContents("ruimtereis03", expectedDataContentRuimtereis03)
+    "ruimtereis04" should behave like bagContents("ruimtereis04", expectedDataContentRuimtereis04)
   }
 
+  beforeAll()
+
+  "allfields" should behave like allfieldsSpec()
```

Comment:
I'm confused. Spec is usually an extension for test _classes_ this is a def, too big according to codacy. It starts with an import like a class. If it ... like a duck ...